web/__init__.py

```diff
@@ -115,9 +232,24 @@ def app_api_files_local():
     return {}
 
 
+@app.get("/reload")
+def reload_webserver():
+    config = app.config["config"]
+    with config.open() as cfg:
+        app.config["config"] = config
+        app.config["login"] = True if cfg else False
+        if cfg:
+            session['_flashes'].clear()
+            flash('Configuration loaded', 'success')
+            startup()
```

Comment:
This is bad - in the current code, this would register all services twice, which will cause an exception.

Register the services always, and just load the config.

I'm not convinced this function/endpoint is needed at all.

Reply:
No, if you look in current code, I have forced startup to not load services if there is no login, so this would ensure services are loaded on the redirect back to `/` so that you don't have to restart the webserver.

Reply:
The services are loaded on-demand. They only start up when they are requested, so the config isn't needed until then.

Also, we can't register the services twice, so this way will crash if we hit `/reload` again.

Reply:
I'm not trying to be rude, but in my testing, this is absolutely not true. In testing, in the `startup()` function, if `app.config["config"]` is not valid when the services load, it causes the app to crash and reboot. Thus I added the `if app.config["Login"]` clause to only start up those services when there is a valid login. Thus calling it in this function is calling it again once there is a valid login so the services then get registered. 

In my code, I have uploaded my config, then re-uploaded my config (which triggers `/reload` a second time) and there was no subsequent crash, the app continued to function as intended. The only ill effect i've seen in testing is that even though we don't load the video section when there is no config, it still attempts to call the pppp and mqtt service endpoints which at this point aren't registered thus triggering a 404 in logs but the end user never sees this.

Reply:
✅